server/src/test/java/org/elasticsearch/cluster/metadata/DesiredNodesTestCase.java

```diff
@@ -44,38 +43,28 @@ public static DesiredNodeWithStatus randomDesiredNodeWithStatus() {
     }
 
     public static DesiredNode randomDesiredNode() {
-        return randomDesiredNode(Version.CURRENT, Settings.EMPTY);
+        return randomDesiredNode(Settings.EMPTY);
     }
 
     public static DesiredNode randomDesiredNode(Settings settings) {
-        return randomDesiredNode(Version.CURRENT, settings);
+        return randomDesiredNode(settings, randomProcessorRange());
```

Comment:
This needs the `randomBoolean()` between random processor ranges & random number of processors

Reply:
True, good spot, thanks!